src/lib/y2partitioner/widgets/fstab_options.rb

```diff
@@ -0,0 +1,564 @@
+        Yast::UI.ChangeWidget(Id(:mt_group), :Value, value)
+      end
+
+      def contents
+        RadioButtonGroup(
+          Id(:mt_group),
+          VBox(
+            Left(Label(label)),
+            HBox(
+              VBox(
+                Left(RadioButton(Id(:device), _("&Device Name"))),
+                Left(RadioButton(Id(:label), _("Volume &Label"))),
+                Left(RadioButton(Id(:uuid), _("&UUID")))
+              ),
+              Top(
+                VBox(
+                  Left(RadioButton(Id(:id), _("Device &ID"))),
+                  Left(RadioButton(Id(:path), _("Device &Path")))
+                )
+              )
+            )
+          )
+        )
+      end
+
+      def selected_mount_by
+        Y2Storage::Filesystems::MountByType.all.detect do |fs|
+          fs.to_sym == value
+        end
+      end
+
+      def value
+        Yast::UI.QueryWidget(Id(:mt_group), :Value)
+      end
+    end
+
+    # A group of options that are general for many filesystem types.
+    class GeneralOptions < CWM::CustomWidget
+      include FstabCommon
+
+      def contents
+        return Empty() unless widgets.any? { |w| draw_widget?(w) }
+
+        VBox(* widgets.map { |w| draw(w) }, VSpacing(1))
+      end
+
+      def widgets
+        [
+          ReadOnly.new(@options),
+          Noatime.new(@options),
+          MountUser.new(@options),
+          Noauto.new(@options),
+          Quota.new(@options)
+        ]
+      end
+    end
+
+    # CheckBox to disable the automount option when starting up
+    class Noauto < CWM::CheckBox
+      include FstabCommon
+
+      VALUES = ["noauto", "auto"].freeze
+
+      def init
+        self.value = @options.fstab_options.include?("noauto")
+      end
+
+      def store
+        delete_from_fstab!(Regexp.union(VALUES))
+
+        @options.fstab_options << "noauto" if value
+      end
+
+      def label
+        _("Do Not Mount at System &Start-up")
+      end
+    end
+
+    # CheckBox to enable the read only option ("ro")
+    class ReadOnly < CWM::CheckBox
+      include FstabCommon
+      VALUES = ["rw", "ro"].freeze
+
+      def init
+        self.value = @options.fstab_options.include?("ro")
+      end
+
+      def store
+        delete_from_fstab!(Regexp.union(VALUES))
+
+        @options.fstab_options << "ro" if value
+      end
+
+      def help
+        "<p><b>Mount Read-Only:</b>\n" \
+        "Writing to the file system is not possible. Default is false. During installation\n" \
+        "the file system is always mounted read-write.</p>"
+      end
+
+      def label
+        _("Mount &Read-Only")
+      end
+    end
+
+    # CheckBox to enable the noatime option
+    class Noatime < CWM::CheckBox
+      include FstabCommon
+      VALUES = ["noatime", "atime"].freeze
+
+      def init
+        self.value = @options.fstab_options.include?("noatime")
+      end
+
+      def store
+        delete_from_fstab!(Regexp.union(VALUES))
```

Comment:
in fact union is not needed due to implementation where "atime" itself is enough to delete also noatime and also potentially other options.